client/my-sites/site-settings/form-writing.jsx

```diff
@@ -57,6 +57,22 @@ const SiteSettingsFormWriting = React.createClass( {
 		return settings;
 	},
 
+	isCustomPostTypesSettingsEnabled() {
+		if ( ! config.isEnabled( 'manage/custom-post-types' ) ) {
+			return false;
+		}
+
+		const { site } = this.props;
+		if ( ! site || ! site.jetpack ) {
+			return true;
+		}
+
+		return (
+			site.versionCompare( '4.2.0', '>=' ) >= 0 &&
```

Comment:
shouldn't this be `site.versionCompare( '4.2.0', '>=' ) &&` - since `versionCompare` returns a boolean?